backend/src/main/java/com/google/sps/servlets/AddParticipantServlet.java

```diff
@@ -56,33 +58,48 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Request parameter values
-    // TODO: Check input values are filled (before allowing to submit)
-    UserService userService = UserServiceFactory.getUserService();
-    String email = userService.getCurrentUser().getEmail();
-    if (email == null) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid email.");
-      return;
+
+    // Retrieve body of HTTP request
+    StringBuffer requestBuffer = new StringBuffer();
+    try {
+      BufferedReader reader = request.getReader();
+      String currentLine;
+      while ((currentLine = reader.readLine()) != null) {
+        requestBuffer.append(currentLine);
+      }
+    } catch (Exception e) {
```

Comment:
When catching exceptions you want to try to catch the most specific exception possible. Catching `Exception` is very broad and can lead to unexpected scenarios.

Reply:
Changed to catching and IOException